libraries/GCS_MAVLink/GCS_FTP.cpp

```diff
@@ -550,8 +550,28 @@ void GCS_MAVLINK::ftp_worker(void) {
                         }
                         break;
                     }
+
+                case FTP_OP::Rename: {
+                    // sanity check that the request looks well formed
+                    const char *filename1 = (char*)request.data;
+                    const size_t len1 = strnlen(filename1, sizeof(request.data)-2);
+                    const char *filename2 = (char*)&request.data[len1+1];
+                    const size_t len2 = strnlen(filename2, sizeof(request.data)-(len1+1));
+                    if (filename1[len1] != 0 || (len1+len2+1 != request.size) || (request.size == 0)) {
```

Comment:
if request.size is 0 won't the access above potentially go to garbage?

Reply:
the packet is always there, even if it was filled with random data and request.size was zero the use of strnlen() means it won't look beyond the end of the packet data. You will get the ftp error